### DISCUSSION TITLE
How do I set VCPKG_ROOT

### DISCUSSION THREAD

**Discussion by igorgsouza** [microsoft/vcpkg#25394] [Q&A] [answered]
I am trying to make a CMake configuration where I dont need to link the vcpkg manually, so I need to use VCPKG_ROOT.

However it seems that this variable cannot be found. I am using a WSL Unbuntu 20.04, do I need to manually set VCPKG_ROOT or is it something that `vcpkg integrate install` should do?

**Comment by nabeelsherazi** [comment, marked as answer]
You would need to manually do this if you're going down this route. On POSIX systems (like WSL), you can set an environment variable for the duration of your current shell (and its subprocesses) with `export VARIABLE=VALUE`. To set this permanently, one way would be edit your `~/.bashrc` file and append `export VCPKG_ROOT=<your path to vcpkg>` to the end. However, it might be better to avoid using a system-wide vcpkg on Linux and use the submodule route instead. You could then set the toolchain file in the first few lines of your CMakeLists.txt with a relative path.

**Reply by igorgsouza** [reply]
Thanks!
